### PR TITLE
Added loading input to infinite scroll

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-infinite-scroll-loading-prop](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-infinite-scroll-loading-prop)

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 1]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
     }
   }
 
+  /**
+   * @required
+   * Specifies that data is loading as a result of the `scrollEnd` event. Setting the property
+   * to `true` disables new `scrollEnd` events from firing until it is set to `false`. If this
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:54]:
```suggestion
   * to `true` disables new `scrollEnd` events from firing until it changes to `false`. If this
```

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 2]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
...
+   * property is not specified, the infinite scroll component will watch the DOM for changes
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:55]:
```suggestion
   * property is not specified, the infinite scroll component watches the DOM for changes
```

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 3]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
...
+   * and begin firing `scrollEnd` events when changes occur on its parent DOM element. Relying
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:56]:
```suggestion
   * and fires `scrollEnd` events when changes occur on its parent DOM element. Relying
```

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 4]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
...
+   * on this behavior could cause the `scrollEnd` event to fire an excessive amount of times
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:57]:
```suggestion
   * on this default behavior could fire an excessive number of `scrollEnd` events
```

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 5]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
...
+   * if the DOM changes are unrelated to loading new data, so it is highly recommend that this
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:58]:
```suggestion
   * if the DOM changes are not related to loading data, so we strongly recommend using this
```

File: src/app/public/modules/infinite-scroll/infinite-scroll.component.ts [hunk 6]
```diff
@@ -48,6 +48,29 @@ export class SkyInfiniteScrollComponent implements OnDestroy {
...
+   * property be used to explicitly set the infinite scroll's loading state.
```
Comment (blackbaud-johnly) [inline src/app/public/modules/infinite-scroll/infinite-scroll.component.ts:59]:
```suggestion
   * property to explicitly set the infinite scroll's loading state.
```

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-infinite-scroll-loading-prop](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-infinite-scroll-loading-prop)

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**